src/ui/DynamicFacetManager/DynamicFacetManager.ts

```diff
@@ -129,7 +129,10 @@ export class DynamicFacetManager extends Component {
     Assert.exists(data);
     Assert.exists(data.queryBuilder);
 
-    this.allFacets.forEach(dynamicFacet => dynamicFacet.putStateIntoQueryBuilder(data.queryBuilder));
+    this.allFacets.forEach(dynamicFacet => {
+      dynamicFacet.putStateIntoQueryBuilder(data.queryBuilder);
```

Comment:
If `putStateIntoQueryBuilder` and `putStateIntoAnalytics` should normally be called one after the other, the code would be easier to use/less error-prone if we had a method that calls both. You could call it here and on line 472 in the above file.

Reply:
Im not sure, they're both pretty independent, and putStateIntoQueryBuilder has already been released + documented (though new doc hasnt been released) so it would be bad pratice to rename it